governance/program/src/instruction.rs

```diff
@@ -0,0 +1,214 @@
+    MAX_PROPOSAL_NAME_LENGTH,
+};
+
+/// Instructions supported by the Governance program
+#[derive(Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum GovernanceInstruction {
+    /// Creates Governance account
+    ///
+    ///   0. `[writable]` Governance account. The account pubkey needs to be set to program-derived address (PDA) with the following seeds:
+    ///           1) 'governance' const prefix,
+    ///           2) Governed Program account key
+    ///   1. `[]` Account of the Program governed by this Governance account
+    ///   2. `[writable]` Program Data account of the Program governed by this Governance account
+    ///   3. `[signer]` Current Upgrade Authority account of the Program governed by this Governance account
+    ///   4. `[]` Governance mint that this Governance uses
+    ///   5. `[signer]` Payer
+    ///   6. `[]` System account
+    ///   7. `[]` Bpf_upgrade_loader account
+    ///   8. `[]` Council mint that this Governance uses [Optional]
+    CreateGovernance {
+        /// Voting threshold in % required to tip the vote
+        /// It's the percentage of tokens out of the entire pool of governance tokens eligible to vote
+        vote_threshold: u8,
+
+        /// Minimum waiting time in slots for a transaction to be executed after proposal is voted on
+        min_transaction_hold_up_time: u64,
+
+        /// Time limit in slots for proposal to be open for voting
+        max_voting_time: u64,
+
+        /// Minimum % of tokens for a governance token owner to be able to create proposal
+        /// It's the percentage of tokens out of the entire pool of governance tokens eligible to vote
+        token_threshold_to_create_proposal: u8,
+
+        /// Governance name
+        name: [u8; MAX_GOVERNANCE_NAME_LENGTH],
+    },
+
+    /// Initializes a new empty Proposal for Instructions that will be executed at various slots in the future
+    /// The instruction also grants Admin and Signatory token to the provided account
+    ///
+    ///   0. `[writable]` Uninitialized Proposal State account
+    ///   1. `[writable]` Uninitialized Proposal account
+    ///   2. `[writable]` Initialized Governance account
+    ///   3. `[writable]` Initialized Signatory Mint account
+    ///   4. `[writable]` Initialized Admin Mint account
+    ///   5. `[writable]` Initialized Admin account for the issued admin token
+    ///   6. `[writable]` Initialized Signatory account for the issued signatory token
+    ///   7. `[writable]` Initialized Source Token Holding account
+    ///   8. `[]` Source mint account
+    ///   9. `[]` Proposal Authority account. PDA with seeds: ['governance',proposal_key]
+    ///   10. '[]` Token program account
+    ///   11. `[]` Rent sysvar
+    InitializeProposal {
+        /// Link to gist explaining proposal
+        description_link: [u8; MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH],
+        /// Name of the proposal
+        name: [u8; MAX_PROPOSAL_NAME_LENGTH],
```

Comment:
How about using the `String` type here?  Passing in these two large arrays means that the `InitializeProposal` instruction data is mostly full of unnecessary zeros.  Also please document the character format (UTF-8 I assume).

Reply:
Types changed to String. 
Should I still add explicit comment for UTF-8  or we consider it to be implied by using the String type please?

Reply:
It wouldn't hurt to be explicit IMO.  The program implementation can use `from_utf8()`, like the memo program, to validate the UTF8 input string:
https://github.com/solana-labs/solana-program-library/blob/b99c9b375c6895a4a94dadd578c3a142b924a6fe/memo/program/src/processor.rs#L28

Reply:
Done